python/adbc_driver_flightsql/adbc_driver_flightsql/__init__.py

```diff
@@ -38,8 +39,22 @@ def connect(
     db_kwargs : dict, optional
         Initial database connection parameters.
     """
-    root = importlib.resources.files(__package__)
-    entrypoint = root.joinpath("libadbc_driver_flightsql.so")
     return adbc_driver_manager.AdbcDatabase(
-        driver=str(entrypoint), uri=uri, **(db_kwargs or {})
+        driver=_driver_path(), uri=uri, **(db_kwargs or {})
     )
+
+
+@functools.cache
+def _driver_path() -> str:
+    root = importlib.resources.files(__package__)
+    entrypoint = root.joinpath("libadbc_driver_flightsql.so")
+    if entrypoint.is_file():
+        return str(entrypoint)
+    is_conda = root.joinpath(".is_conda")
+    if is_conda.is_file():
+        with is_conda.open() as source:
+            return source.read().strip()
+    raise RuntimeError(f"Could not find driver, was {__package__} properly installed?")
```

Comment:
Should we try to make this less conda-specific? I would assume also other package managers will want to do the same as conda (it's actually wheels that's the special case here, I suppose we will only need to look into the package itself in case of wheels?)

The logic that I have (seen) used in other packages with vendored data/libs in wheels: check if the vendored lib file exists, if so use that, otherwise search in a standard location based on sys.prefix.

Reply:
This will then also work for local development if you let install the c driver in the standard install prefix

Reply:
Fair point. I'll switch this to search something like `sys.prefix + '/lib'` which appears to be `$CONDA_PREFIX` or `/usr` on my Linux machine at least.